R/SupervisedTask.R

```diff
@@ -1,13 +1,13 @@
-makeSupervisedTask = function(type, data, target, weights, blocking, fixup.data, check.data) {
-  task = makeTask(type = type, data = data, weights = weights, blocking = blocking, fixup.data = fixup.data, check.data = check.data)
+makeSupervisedTask = function(type, data, target, weights, blocking, fixup.data, check.data, spatial) {
+  task = makeTask(type = type, data = data, weights = weights, blocking = blocking, fixup.data = fixup.data, check.data = check.data, spatial = spatial)
 
   if (check.data) {
     # costsens does not have a target col...
     # assertCharacter(target, any.missing = FALSE, min.len = 1L)
     w = which.first(target %nin% colnames(data))
     if (length(w) > 0L)
       stopf("Column names of data doesn't contain target var: %s", target[w])
-    checkTaskData(task$env$data, cols = setdiff(colnames(data), target))
+    checkTaskData(task$env$data, cols = setdiff(colnames(task$env$data), target))
```

Comment:
If you modify the task, the other stuff the user wants to do with it will potentially break. Yes, there's a bit more code, but I don't think it's actually that bad. Documenting this isn't enough, it's completely unexpected behaviour -- why should building a predictive model change the underlying data?